server/user.js

```diff
@@ -0,0 +1,104 @@
+/**
+ * @fileoverview user function handler
+ */
+var fs = require('fs');
+var childProcess = require('child_process');
+var readline = require('readline');
+var mkdirp = require('mkdirp');
+
+var utils = require('./utils.js');
+
+/** @type {user} */
+module.exports = user;
+
+/**
+ * @constructor
+ */
+function user() {}
+
+/** @enum {string} */
+user.QueryType = {
+  SIGHUP: 'sign-up',
+  SIGNIN: 'sign-in'
+};
+
+/**
+ * Name of user information file.
+ * @type {string}
+ */
+user.userInfoFile = 'userInfo.txt';
+
+/**
+ * Checks the user information for signing in.
+ * @param {string} userPath Directory of user indormation file.
+ * @param {!Object} userInfo User Information
+ * @return {!Object{
```

Comment:
Hmm. It's just that we decided it could be cleaner to handle errors by replying http error status code (e.g. 500), so that the front-end could capture that by jquery premise. Otherwise the front-end success handler has to diferentiate success from failure and it makes the handler code a bit messy, e.g.
```js
$.get()
  .done(function(data) {
  if(data.error) { 
   ...
  } else {
   ...
   } 
});
```
See how server/network.js is handling file reading errors. We will gradually switch to the new handling method.
```js
$.get()
  .done(...)
  .fail(...); // error capturing premise
```